plotter/volcano

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+cat >volcano_example.go <<EOF
+// Generated code do not edit. Run \`go generate\`.
+
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build ignore
+
+//go generate ./volcano
+
+package main
+
+import (
+	"github.com/gonum/matrix/mat64"
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/palette"
+	"github.com/gonum/plot/plotter"
+)
+
+type deciGrid struct{ mat64.Matrix }
+
+func (g deciGrid) Dims() (c, r int)   { r, c = g.Matrix.Dims(); return c, r }
+func (g deciGrid) Z(c, r int) float64 { return g.Matrix.At(r, c) }
+func (g deciGrid) X(c int) float64 {
+	_, n := g.Matrix.Dims()
+	if c < 0 || c >= n {
+		panic("index out of range")
+	}
+	return 10 * float64(c)
+}
+func (g deciGrid) Y(r int) float64 {
+	m, _ := g.Matrix.Dims()
+	if r < 0 || r >= m {
+		panic("index out of range")
+	}
+	return 10 * float64(r)
+}
+
+func main() {
+	var levels []float64
+	for l := 100.5; l < volcano.Matrix.(*mat64.Dense).Max(); l += 5 {
+		levels = append(levels, l)
+	}
+	c := plotter.NewContour(volcano, levels, palette.Rainbow(len(levels), (palette.Yellow+palette.Red)/2, palette.Blue, 1, 1, 1))
+	h := plotter.NewHeatMap(volcano, palette.Heat(len(levels)*2, 1))
+
+	p, err := plot.New()
+	if err != nil {
+		panic(err)
+	}
+	p.Title.Text = "Maunga Whau Volcano"
+
+	p.Add(h)
+	p.Add(c)
+
+	p.X.Padding = 0
+	p.Y.Padding = 0
+	_, p.X.Max, _, p.Y.Max = h.DataRange()
+
+	name := "example_volcano"
+
+	for _, ext := range []string{
+		".eps",
+		".pdf",
+		".svg",
+		".png",
+		".tiff",
+		".jpg",
+	} {
+		if err := p.Save(4, 4, name+ext); err != nil {
+			panic(err)
+		}
+	}
+}
+
+// Data extracted from RDatasets volcano data for the Maunga Whau volcano topographic data.
+var volcano = deciGrid{mat64.NewDense(87, 61, []float64{
+EOF
+R -q -e 'write.table(as.data.frame(volcano), file="volcano_example.go", sep=", ", eol=",\n", col.names=FALSE, row.names=FALSE, append=TRUE)'
```

Comment:
Do I need R installed to run this example?

Can we have a different example? I feel like this is an unidiomatic use of 'go generate'. In general, I don't think go generate is intended to be used by library users, instead just by authors. Instead, could we establish something like gonum/plot/examples/examplename where they are all compileable examples? I don't know what the licensing is on the volcano data, but we could have the data marshaled into a *mat64.Dense directly. It's fine to use generate to do that construction (which is what it looks like this execution does), but the actual published volcano_example should be able to be compiled and ran without using go generate.